sklearn/manifold/_isomap.py

```diff
@@ -297,6 +298,7 @@ def _fit_transform(self, X):
                 **self.nbrs_.effective_metric_params_,
             )
 
+        _ensure_sparse_index_int32(nbg)
```

Comment:
How did you run into this issue? (And know that you need to ensure int32)

Reply:
Many tests fail in `manifold/tests/test_isomap.py` with SciPy v1.12. For example `sklearn/manifold/tests/test_isomap.py::test_isomap_simple_grid`.  
They do pass under SciPy 1.8 and SciPy 1.16. I ran the test suite for SciPy v1.8, v1.12 and v1.16 on this branch. And the v1.12 environment caused these failures.

I know that recent SciPy versions (I think 1.14+) already ensure index arrays are `int32` because I added that. But I don't know the history for why that isn't needed for SciPy v1.8.